nucypher/blockchain/eth/agents.py

```diff
@@ -150,14 +150,18 @@ def approve_transfer(self, amount: int, target_address: str, sender_address: str
 
     def transfer(self, amount: int, target_address: str, sender_address: str, auto_approve: bool = True):
         if auto_approve:
+            
             allowance = self.contract.functions.allowance(sender_address, target_address).call()
-            if allowance != 0:
+            
+            if allowance == 0: 
+                self.approve_transfer(amount=amount, target_address=target_address, sender_address=sender_address)
```

Comment:
Was this line causing you an Issue?

Reply:
remember last week, it was causing an error when we 1st turned the node on.

Reply:
FWIW - this is deprecated in #1225 - This entire `approve_transfer` and `increase_allowance` block are improperly implemented here.